mtf/metadata/tmet/common.py

```diff
@@ -0,0 +1,291 @@
+
+    def logic_filter(actual_tag_list, tag_filter):
+        # TODO: try to replace this part with http://www.sympy.org
+        statement_or = False
+        # if no tags in test then add this test to set (same behaviour as avocado --tag...empty)
+        # print actual_tag_list, tag_filter
+        if not actual_tag_list:
+            statement_or = True
+        actualinput = logic_and(actual_tag_list)
+        for onefilter in tag_filter:
+            filterinput = logic_and(onefilter)
+            statement_and = True
+            for key in filterinput:
+                if filterinput.get(key) != bool(actualinput.get(key)):
+                    statement_and = False
+                    break
+            if statement_and:
+                statement_or = True
+                break
+        return statement_or
+
+    if op_or:
+        filters = filters.split(op_or)
+    return logic_filter(statement, filters)
+
+
+class MetadataLoader(object):
+    base_element = {}
+    backends = ["generic", "general"]
+    # in filter there will be items like: {"relevancy": None, "tags": None}
+    filter_list = [None]
+
+    def __init__(self, location=".", linters=False, **kwargs):
+        self.location = os.path.abspath(location)
+        self._load_recursive()
+        if linters or self.base_element.get(MODULELINT):
+            self._import_linters()
+        if self.base_element.get(IMPORT_TESTS):
+            for testglob in self.base_element.get(IMPORT_TESTS):
+                self._import_tests(os.path.join(self.location, testglob))
+        if self.base_element.get(TAG_FILETERS):
+            self.add_filter(tags=self.base_element.get(TAG_FILETERS))
+
+    def _import_tests(self, testglob, pathlenght=0):
+        pathglob = testglob if testglob.startswith(os.pathsep) else os.path.join(self.location, testglob)
+        print_debug("Import tests: %s" % pathglob)
+        for testfile in glob.glob(pathglob):
+            test = {SOURCE: testfile, DESC: "Imported tests by path: %s" % pathglob}
+            self._insert_to_test_tree(testfile.strip(os.sep).split(os.sep)[pathlenght:],
+                                      test)
+
+    def _import_linters(self):
+        raise NotImplementedError
+
+    def get_metadata(self):
+        return self.base_element
+
+    def load_yaml(self, location):
+        print_debug("Loading metadata from file: %s" % location)
+        with open(location, 'r') as ymlfile:
+            xcfg = yaml.load(ymlfile.read())
+        if xcfg.get(DOCUMENT[0]) not in DOCUMENT[1]:
+            raise BaseException("bad yaml file: item (%s)", xcfg.get(DOCUMENT[0]))
+        else:
+            return xcfg
+
+    def _insert_to_coverage(self, path_list, test):
+        coverage_key = "/".join(path_list)
+        print_debug("insert to coverage %s to %s" % (test, coverage_key))
+        # add test coverage key
+        # add backend key if does not exist
+        self.base_element[TESTC][coverage_key] = test
```

Comment:
if what you describe is alias for my code, then yes, but I'm not sure if this is same.
eg: TESTC is already here and contain some keys, this will rewrite it just with `{coverage_key: test}` but I'm not sure